ui/dashboards/src/context/TimeRangeProvider.tsx

```diff
@@ -44,34 +45,24 @@ export function TimeRangeProvider(props: TimeRangeProviderProps) {
         return;
       }
 
+      // needed for TimeRangeControls since absolute time calendar and relative time shortcuts supported
+      setSelectedTimeRange(value);
+
+      // convert to absolute time range if relative time shortcut passed
       if (isRelativeTimeRange(value)) {
-        if (setQueryString) {
-          queryString.set('start', value.pastDuration);
-          // end not required for relative time but may have been set by AbsoluteTimePicker or zoom
-          queryString.delete('end');
-          setQueryString(queryString);
-        } else {
-          setActiveTimeRange(toAbsoluteTimeRange(value));
-        }
+        setActiveTimeRange(toAbsoluteTimeRange(value));
```

Comment:
Does this really need to be a separate piece of state? Looks like it can potentially just be "derived" maybe?

Reply:
This is fixed now 🙌 (after refactoring useTimeRange, see other [comment here](https://github.com/perses/perses/pull/621#discussion_r993889578)). 

I also adjusted TimeRangeProvider's props like we talked about: https://github.com/perses/perses/blob/1b5377230d8a9cc04796019f4c084dc3e574a0ce/ui/dashboards/src/views/ViewDashboard/ViewDashboard.tsx#L40-L45